### PR TITLE
Removing classifyIntent call to LLM to optimize Fixup recipes

### PR DESCRIPTION
**Pull request by arafatkatze**
Solves Part 1 of #1544 
Quoting @umpox 
image

## Test plan

Tested locally in debug mode with fixup/chat/inline

### CODE REVIEW

File: lib/shared/src/intent-detector/client.ts [hunk 1]
```diff
@@ -1,17 +1,11 @@
-import { ANSWER_TOKENS } from '../prompt/constants'
-import { Message } from '../sourcegraph-api'
-import { SourcegraphCompletionsClient } from '../sourcegraph-api/completions/client'
 import { SourcegraphGraphQLAPIClient } from '../sourcegraph-api/graphql'
```
Comment (umpox) [inline lib/shared/src/intent-detector/client.ts:1]: @arafatkatze Let's not remove the changes in this file.

The intent detector logic is generic enough as-is and we'll likely find value for this in the near future.

File: lib/shared/src/intent-detector/index.ts [hunk 2]
```diff
@@ -26,9 +26,4 @@ export interface IntentClassificationOption<Intent = string> {
 export interface IntentDetector {
     isCodebaseContextRequired(input: string): Promise<boolean | Error>
     isEditorContextRequired(input: string): boolean | Error
-    classifyIntentFromOptions<Intent extends string>(
-        input: string,
-        options: IntentClassificationOption<Intent>[],
-        fallback: Intent
-    ): Promise<Intent>
```
Comment (umpox) [inline lib/shared/src/intent-detector/index.ts:33]: Keep this too

File: vscode/src/non-stop/FixupController.ts [hunk 3]
```diff
@@ -217,12 +210,7 @@ export class FixupController
             return 'add'
         }
 
-        const intent = await intentDetector.classifyIntentFromOptions(
-            task.instruction,
-            FixupIntentClassification,
-            'edit'
-        )
-        return intent
+        return 'edit'
```
Comment (umpox) [inline vscode/src/non-stop/FixupController.ts:213]: 👍

**Review by umpox — APPROVED** [review]
Looks good, thanks @arafatkatze

Left a few comments to address before we merge!

**Comment by arafatkatze** [comment]
@umpox Thanks for the comments I made the suggested changes.

**Comment by umpox** [comment]
CI all green, will merge - thank you @arafatkatze!

**Pull request merged**